src/util/taproot.rs

```diff
@@ -82,6 +84,31 @@ sha256t_hash_newtype!(TapSighashHash, TapSighashTag, MIDSTATE_TAPSIGHASH, 64,
     doc="Taproot-tagged hash for the taproot signature hash", true
 );
 
+impl From<schnorr::PublicKey> for TapTweakHash {
+    #[inline]
+    fn from(pubkey: schnorr::PublicKey) -> Self {
+        TapTweakHash::hash(&pubkey.serialize())
+    }
+}
+
+impl TapTweakHash {
+    /// Generates [`TapTweakHash`] from a given BIP-340 key pair ([`schnorr::KeyPair`])
+    #[inline]
+    pub fn with<C: Signing>(secp: &Secp256k1<C>, pair: &schnorr::KeyPair) -> Self {
+        let pubkey = schnorr::PublicKey::from_keypair(&secp, pair);
+        TapTweakHash::hash(&pubkey.serialize())
+    }
+}
+
+/// Trait for enabling tweaking methods to BIP-340 keys according to the Taroot rules
+pub trait TaprootKey {
```

Comment:
Is there any reason for implementing this for Keypair? I think if we wait for #677 to be merged, this PR can be made simpler.

Reply:
agree to re-base on #677